.github/workflows/pre-merge-checks.yml

```
@@ -6,84 +6,78 @@ on:
       DEV_CRI_CIC_PRE_MERGE_ROLE_ARN:
         required: true
 
+concurrency:
+  group: pre-merge-check${{ github.head_ref || github.ref_name }}
+  cancel-in-progress: true
+      
 jobs:
   setup:
     runs-on: ubuntu-latest
     timeout-minutes: 15
-    concurrency:
-      group: build-pre-merge-${{ github.head_ref || github.ref_name }}
     outputs:
       sha_short: ${{ steps.vars.outputs.sha_short }}
-    defaults:
-      run:
-        working-directory: ./deploy/
     steps:
       - name: Check out repository code
         uses: actions/checkout@v5
-        with:
-          fetch-depth: 0
 
       - name: Get output variables
         id: vars
         run: |
           echo "sha_short=$(git rev-parse --short HEAD)" >> $GITHUB_OUTPUT
 
-      - name: Setup Node.js
-        uses: actions/setup-node@v5
-        with:
-          node-version: 22
-          cache: 'npm'
-          cache-dependency-path: 'src/package-lock.json'
-
-      - name: Install dependencies
-        run: cd ../src && npm ci
-
-      - name: Setup SAM CLI
-        uses: aws-actions/setup-sam@v2
-        with:
-          use-installer: true
-
-      - name: SAM Build
-        run: sam build -t template.yaml
-
-      - name: Cache SAM build
-        uses: actions/cache/save@v4
-        with:
-          path: deploy/.aws-sam
-          key: sam-build-${{ github.sha }}
-
   deploy:
     runs-on: ubuntu-latest
     timeout-minutes: 15
     needs: setup
-    concurrency:
-      group: deploy-pre-merge-${{ github.head_ref || github.ref_name }}
     permissions:
       id-token: write
       contents: read
+    outputs:
+        api_url: ${{ steps.api_url.outputs.api_url }}
     steps:
       - name: Check out repository code
         uses: actions/checkout@v5
 
+      - name: Setup SAM CLI
+        uses: aws-actions/setup-sam@v2
+        with:
+          use-installer: true
+
       - name: Assume temporary AWS role
         uses: aws-actions/configure-aws-credentials@v4
         with:
           role-to-assume: ${{ secrets.DEV_CRI_CIC_PRE_MERGE_ROLE_ARN }}
           aws-region: eu-west-2
 
+      - name: SAM Build API stack
+        working-directory: ./deploy
+        run: sam build -t template.yaml
+
+      - name: SAM Deploy API stack
+        working-directory: ./deploy
+        run: |
+          sam deploy \
+          --stack-name "pre-merge-${{ needs.setup.outputs.sha_short }}" \
+          --s3-bucket ${{secrets.DEV_CRI_CIC_ARTIFACT_SOURCE_BUCKET_NAME}} \
+          --template-file .aws-sam/build/template.yaml \
+          --capabilities CAPABILITY_NAMED_IAM \
+          --parameter-overrides Environment=dev
+      
+      - name: Construct API URL
+        id: api_url
+        run: |
+          echo "api_url=https://api-pre-merge-${{ needs.setup.outputs.sha_short }}-review-o-dev.account.gov.uk" >> $GITHUB_OUTPUT 
```

Minor error with the api url 
```suggestion
          echo "api_url=https://api-pre-merge-${{ needs.setup.outputs.sha_short }}.review-c.dev.account.gov.uk" >> $GITHUB_OUTPUT 
```